src/commands/aksInspektorGadget/aksInspektorGadget.ts

```diff
@@ -80,6 +135,27 @@ async function prepareInspektorGadgetInstall(
                 await unDeployGadget(clustername, kubeconfig, kubectl);
             }
             return;
+        case Command.TopTCP:
+            await topTCPGadget(clustername, kubeconfig, kubectl);
+            return;
+        case Command.TopEBPF:
+            await topEBPFGadget(clustername, kubeconfig, kubectl);
+            return;
+        case Command.TopBlockIO:
+            await topBlockIOGadget(clustername, kubeconfig, kubectl);
+            return;
+        case Command.TopFile:
+            await topFile(clustername, kubeconfig, kubectl);
+            return;
+        case Command.ProfileCPU:
+            await profileCPU(clustername, kubeconfig, kubectl);
+            return;
+        case Command.SnapshotProcess:
+            await snapshotProcess(clustername, kubeconfig, kubectl);
+            return;
+        case Command.SnapshotSocket:
+            await snapshotSocket(clustername, kubeconfig, kubectl);
+            return;
```

Comment:
Currently, if we try to run one of these gadgets without having deployed Inspektor Gadget before, it will fails with error: 

```
kubectl gadget command failed with following error: Kubectl returned error 1 for gadget snapshot socket --all-namespaces 30 --timeout 30
Error: Error: failed to run gadget: error creating trace: no gadget pods found. Is Inspektor Gadget deployed?
```

What if we verify that Inspektor Gadget is running and in case it is not, we ask the users if they want to deploy it? To verify it is running, you can use the `version` command:

```bash
# Not running
$ ./kubectl-gadget version
Client version: v0.12.1
Server version: not installed

# ./kubectl-gadget deploy
[...]

# Running
$ ./kubectl-gadget version
Client version: v0.12.1
Server version: v0.12.1
```

BTW, we could improve `version` command to allow using JSON format so that it is easier to parse:

```bash
$ ./kubectl-gadget version -o json
{"Client version": "v0.12.1", "Server version": "v0.12.1"}
```

Reply:
@blanquicet - that's a great point, and definitely something we should be looking at.

I think a more general point is that the tree-view/command interface doesn't quite capture the state of the various resources deployed to the cluster. In an ideal world, only commands which make sense should be displayed to the users, so it'd be preferable if only 'deploy' were available in the case where IG is not deployed.

I think it'd be _possible_ to selectively display commands based on querying the state of the cluster, but it'd require running background queries to poll all the clusters in the explorer view, which is a performance trade-off we've been very wary of making.

An alternative approach that @Tatsinnit and I have been chatting about is to use a webview. There are possible accessibility concerns here, which will need a bit more investigation. But it could solve a few issues that have been raised so far.

The idea (at least in my head) is, you'd have a single 'Inspektor Gadget' command for the cluster, which would launch a webview containing a single button to deploy it (if it's not already deployed). Once deployed, it could have the options to run the various commands, along with whatever filtering inputs we wanted. I think it could support dynamic commands (like 'top') where the output is periodically updated. And there may be a neat way of managing traces, such that you could create multiple long-running ones, and see them listed and interact with them, even if you close and re-open the webview (because they're actual k8s resources, right?).

Reply:
> I think a more general point is that the tree-view/command interface doesn't quite capture the state of the various resources deployed to the cluster. In an ideal world, only commands which make sense should be displayed to the users, so it'd be preferable if only 'deploy' were available in the case where IG is not deployed.
> 
I got your point, but instead of not displaying them at all, I would make them "non-clickable" (grayed out). Otherwise, the users will have to deploy IG just to know what gadgets are available. I think that, sometimes, users could be just curious of exploring what options a new feature offers without installing it.

> The idea (at least in my head) is, you'd have a single 'Inspektor Gadget' command for the cluster, which would launch a webview containing a single button to deploy it (if it's not already deployed). Once deployed, it could have the options to run the various commands, along with whatever filtering inputs we wanted. I think it could support dynamic commands (like 'top') where the output is periodically updated.
>
I like this idea. I would give us more control on what the users can and can't do.

> And there may be a neat way of managing traces, such that you could create multiple long-running ones, and see them listed and interact with them, even if you close and re-open the webview (because they're actual k8s resources, right?).
>
We offer this functionality but it is managed by the `kubectl-gadget` CLI. It mainly translate the CLI arguments in a k8s resource (trace CR) to start tracing with the given set of paramters, but it will also delete such resource when you close the CLI. However, you could not use the CLI and manage the trace CR yourself to start/stop tracing, but of course, it is more complex.

Reply:
Thanks @blanquicet! I've tried to capture this [here](#203), because it's a bit of a bigger conversation than this PR. Please add to it if I've missed anything, or you'd like to add your own ideas.